### DISCUSSION TITLE
Create Static Pages from Database - Querying Content From Database Only on Deploy - Possible?

### DISCUSSION THREAD

**Discussion by hatchli** [vercel/vercel#4139] [General] [answered]
Is it possible to have an app query a database _only_ upon deployment to determine static values?

**Use Case**
 I want a static landing page. However, I also want to to be able to change the landing page if the data in my database changes and then I redeploy. 

So, I don't want the database being queried every time a unique user hits the landing page. But, I want to query the database _only_ upon deploy, and set the header h1 on the landing page to whatever is _at the moment of deploy_ specified in the database to be the header h1.

What I'm imagining right now is having an admin page where the Admin can update the database via the website, and then using the new Deploy Hooks trigger a redeployment. At redeployment, the new values will be queried to populate the static landing page.

Is this possible? 

I'm using React, Apollo Client 3.0, Prisma 2 (so GraphQL), Next.js and a mysql server.

**Comment by coetry** [comment, marked as answer]
If you're using Next.js, do you think [`getStaticProps`](https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation) can solve your use case? It's a function that gets run at build-time, so every-time you deploy, you can query your DB and populate your pages. 

We also have an [RFC](https://github.com/zeit/next.js/discussions/11552) open that will allow you to re-build your pages behind the scenes after a revalidation interval has exceeded, which would update your pages without requiring re-building the entire site.

**Comment by hatchli** [comment]
After doing more research earlier, I had begun down the path of getStaticProps infact! I was initially put off because of an issue preventing me from updating my Next.js package to 9.3, however they released a canary release that works - so that's what I'm doing! Everything's working, and now I can look into RFC! Which is thanks to you @coetry ! Thanks (again) for the help!
